### PR TITLE
commander preflight ekf2 check estimator_status

### PR DESCRIPTION
**Pull request by dagar**
ekf2Check was running regardless of an estimator_status update and not initializing the struct.

### CODE REVIEW

**Comment by dagar** [comment]
I took another pass to simplify a bit more and added a 20s delay before the EKF GPS check is enforced.

**Comment by dagar** [comment]
Rebased and tested.

**Comment by dagar** [comment]
@priseborough could you review the preflight ekf2Check change?https://github.com/PX4/Firmware/pull/7764/files#diff-a2bcb238e25d8a97ab2aee48f54dbc48R747

Many users get an initial "PREFLIGHT FAIL: EKF NOT USING GPS" message which is off putting.

I'm certainly open to a better way to do this rather than the hard coded magic number.

**Comment by priseborough** [comment]
Given the GPS TTFF is highly variable and up to 30 seconds for a cold statrt, and the minimum time required  required for EKF GPS checks to pass after that is 10 seconds and typically 20, we should use the last time gnssCheck was false as the time reference instead of time from boot. 

Longer term we can also improve the reporting if it does fail to indicate the type of GPS quality problem.

I will submit a patch for the timer change.

**Comment by priseborough** [comment]
See https://github.com/priseborough/Firmware/tree/pr7764 for timer reference change.

**Comment by dagar** [comment]
Thanks, I've pulled your commit into this PR.

File: src/modules/commander/PreflightCheck.cpp [hunk 1]
```diff
@@ -562,15 +563,15 @@ static bool ekf2Check(orb_advert_t *mavlink_log_pub, bool optional, bool report_
...
 }
 
-bool preflightCheck(orb_advert_t *mavlink_log_pub, bool checkMag, bool checkAcc, bool checkGyro,
-		    bool checkBaro, bool checkAirspeed, bool checkRC, bool checkGNSS,
+bool preflightCheck(orb_advert_t *mavlink_log_pub, bool checkSensors, bool checkAirspeed, bool checkRC, bool checkGNSS,
 		    bool checkDynamic, bool isVTOL, bool reportFailures, bool prearm, hrt_abstime time_since_boot)
 {
 
-	if (time_since_boot < 1e6) {
+	if (time_since_boot < 2000000) {
```
Comment (potaito) [inline src/modules/commander/PreflightCheck.cpp]: 1. In the PR it says something about 20 seconds, but `2000000` is actually 2 seconds (2.000.000). 
2. Shouldn't this check go into the `airspeedCheck()` function?

**Comment by potaito** [comment]
@dagar I am running into problems with the airspeed driver grace period
https://github.com/PX4/Firmware/blob/18b83f2960879a71d19b456f5823d0abb2950290/src/modules/commander/PreflightCheck.cpp#L612

in my case the drone is booting too fast, runs the preflight check here 
https://github.com/PX4/Firmware/blob/18b83f2960879a71d19b456f5823d0abb2950290/src/modules/commander/commander.cpp#L1621 which already sets `status_flags.condition_system_sensors_initialized` to true. 

I don't actually understand the commander that well and why that `2000000` period needs to be there in the first place, so I can't fix it. Any ideas how to solve this? Should I open an issue? It is clear I think that simply reducing this value again is not a stable solution.

**Comment by dagar** [comment]
It's a grace period waiting for the system to come up. If this is causing trouble please open an issue and we'll go over it in detail.

**Pull request merged**